src/Tablebot/Plugin/SmartCommand.hs

```diff
@@ -144,6 +157,9 @@ instance {-# OVERLAPPABLE #-} (Integral a, Read a) => CanParse a where
 instance CanParse Double where
   pars = double
 
+instance CanParse () where
+  pars = eof
+
```

Comment:
Why does this instance exist? I think all SmartCommands currently assume eof after their final argument? (since the parser generated for `Message -> DatabaseDiscord ()` is just `eof`.) Will come back to this if later code proves me wrong.

Reply:
It was to represent a bare prefix inside an `Either` (i.e. `!quote` when `!quote add` exists). It could probably be more explicit, although subcommanding solves this too.